src/main/java/com/squareup/kotlinpoet/CodeWriter.kt

```diff
@@ -317,7 +317,7 @@ internal class CodeWriter @JvmOverloads constructor(
     if (o is TypeSpec) {
       o.emit(this, null)
     } else if (o is AnnotationSpec) {
-      o.emit(this, true)
+      o.emit(this, inline = true, asParameter = true)
```

Comment:
Hmm I'm a bit unsure about this behavior as it could affect code in places we don't expect. As we don't have any tests which currently exercise this behavior I'm going to opt for the fix instead of being overly defensive. We may need to revisit this should a use-case come up.